iree/compiler/Dialect/HAL/Target/LLVM/StaticLibraryGenerator.cpp

```diff
@@ -0,0 +1,129 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <algorithm>
+#include <cstring>
+#include <fstream>
+#include <iomanip>
+#include <iostream>
+
+#include "llvm/Support/FileSystem.h"
+#include "llvm/Support/Path.h"
+#include "llvm/Support/raw_ostream.h"
+
+namespace mlir {
+namespace iree_compiler {
+namespace IREE {
+namespace HAL {
+
+namespace {
```

Comment:
style: LLVM convention is generally to not have the anonymous namespaces and instead mark the functions as static (though we kind of do both sometimes, which is messy) - the readability improvement there is that you no longer have to order your functions by what is anonymous and not and can keep utility functions closer to where they are used - along with being able to instantly see that a function is local instead of having to scan up to see if you are in an anonymous namespace

Reply:
removed anon namespace.